app/src/test/java/com/kickstarter/viewmodels/CampaignDetailsViewModelTest.kt

```diff
@@ -3,24 +3,98 @@ package com.kickstarter.viewmodels
 import android.content.Intent
 import com.kickstarter.KSRobolectricTestCase
 import com.kickstarter.libs.Environment
+import com.kickstarter.libs.RefTag
+import com.kickstarter.libs.models.OptimizelyExperiment
+import com.kickstarter.mock.MockExperimentsClientType
+import com.kickstarter.mock.factories.ProjectDataFactory
 import com.kickstarter.mock.factories.ProjectFactory
 import com.kickstarter.models.Project
+import com.kickstarter.models.User
 import com.kickstarter.ui.IntentKey
+import com.kickstarter.ui.data.ProjectData
 import org.junit.Test
 import rx.observers.TestSubscriber
 
 class CampaignDetailsViewModelTest : KSRobolectricTestCase() {
     private lateinit var vm: CampaignDetailsViewModel.ViewModel
 
+    private val goBackToProject = TestSubscriber<Void>()
+    private val pledgeContainerIsVisible = TestSubscriber<Boolean>()
     private val url = TestSubscriber<String>()
 
-    private fun setUpEnvironment(environment: Environment, project: Project) {
+    private fun setUpEnvironment(environment: Environment, projectData: ProjectData) {
         this.vm = CampaignDetailsViewModel.ViewModel(environment)
 
+        this.vm.outputs.goBackToProject().subscribe(this.goBackToProject)
+        this.vm.outputs.pledgeContainerIsVisible().subscribe(this.pledgeContainerIsVisible)
         this.vm.outputs.url().subscribe(this.url)
 
-        // Configure the view model with a project.
-        this.vm.intent(Intent().putExtra(IntentKey.PROJECT, project))
+        // Configure the view model with ProjectData.
+        this.vm.intent(Intent().putExtra(IntentKey.PROJECT_DATA, projectData))
+    }
+
+    @Test
+    fun testGoBackToProject() {
+        setUpEnvironment(environment(), ProjectDataFactory.project(ProjectFactory.project()))
+
+        this.vm.inputs.pledgeActionButtonClicked()
+        this.goBackToProject.assertValueCount(1)
+    }
+
+    @Test
+    fun testPledgeContainerIsVisible_whenProjectIsLiveNotBacked_control() {
```

Comment:
Couple questions about these tests:

1) This test appears to be testing the container visibility but asserts after `this.vm.inputs.pledgeActionButtonClicked()`, does the button click affect its visibility?
2) What do you think about `testPledgeContainerIsNotVisible` or `testPledgeContainerVisibility`? Usually I read the test name almost as an assertion that it's testing that _this thing is this_, so in this case I initially thought it was testing that the container _is_ visible whereas it's actually the opposite?

Reply:
1. No, it's copy pasta I need to update 😅 
2. We've been trying to name tests `testOutputName_whenCondition`